### PR TITLE
refactor plugins: output related codes

### LINKED ISSUE #832: Rewrite OUTPUT_JSON printouts to use json-c library
Right now, most of JSON printouts is glued manually using `printf`, which is very prone to syntax errors.

`OUTPUT_JSON` printouts should be rewritten to use json-c which is both more readable and simpler to use, e.g.:
https://github.com/tklengyel/drakvuf/blob/f7fa4b84899a1f5b830c73ac3f50398421011e88/src/plugins/memdump/memdump.cpp#L136

### PR DESCRIPTION
**Pull request by manorit2001**
fixes #832

### CODE REVIEW

**Comment by manorit2001** [comment]
Can we convert `GString*` to `std::string` by some method? I haven't worked with `Gstrings` before

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/plugins/apimon/apimon.cpp [hunk 1]
```diff
@@ -278,6 +278,7 @@ static void print_addresses(drakvuf_t drakvuf, apimon* plugin, const dll_view_t*
             json_object_object_add(j_rvas, target.target_name.c_str(), json_object_new_int(target.offset));
     }
 
+    // does this need to be converted to fmt?
```
Comment (tklengyel) [inline src/plugins/apimon/apimon.cpp:281]: Looks like it

Comment (manorit2001) [reply src/plugins/apimon/apimon.cpp:281]: Ah, I just checked. We don't have `drakvuf_trap_info* info` in the parameter. So, `fmt` can't be used ig

Comment (tklengyel) [reply src/plugins/apimon/apimon.cpp:281]: In that case please remove the added comment

Comment (manorit2001) [reply src/plugins/apimon/apimon.cpp]: Done

**Comment by tklengyel** [comment]
Thanks!

**Pull request merged**